### PR TITLE
Fix wrong genesis block save

### PR DESCRIPTION
**Pull request by iulianpascalau**
- removed a wrong save of the genesis block in hash-nonce storage

Manual testing procedure:
- Start a testnet with nodes that do shuffling between shards. 
- Without the fix, we are able to see an error message containing the string `shard header (nonce-hash)` after a few epochs.
- With the fix, the error message will no longer be logged.

### CODE REVIEW

File: cmd/node/factory/structs.go [hunk 1]
```diff
@@ -577,11 +577,6 @@ func prepareGenesisBlock(args *processComponentsFactoryArgs, genesisBlocks map[u
 		if errNotCritical != nil {
 			log.Error("error storing genesis shardblock", "error", errNotCritical.Error())
 		}
-		hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(genesisBlock.GetShardID())
```
Comment (SebastianMarian) [inline cmd/node/factory/structs.go:580]: Actually this storage exists in each shard but only for self shard. In metachain it is created for every shard.

Comment (iulianpascalau) [reply cmd/node/factory/structs.go:580]: But we do not need the shard genesis block stored inside the nonce-hash unit

Comment (SebastianMarian) [reply cmd/node/factory/structs.go:580]: Ok, in this case you can move line 565 inside first if after line 570

Comment (iulianpascalau) [reply cmd/node/factory/structs.go:580]: done

**Pull request merged**